### PR TITLE
add navbar

### LINKED ISSUE #79: Nav bar component
Create Nav bar component

### PR DESCRIPTION
**Pull request by Mahmoud-Ahmad2**
Relates #79 
Close #79

### CODE REVIEW

**Comment by Mahmoud-Ahmad2** [comment]
Screen Shot 2022-05-19 at 3 42 26 PM

File: client/src/App.js [hunk 1]
```diff
@@ -2,10 +2,10 @@ import React, { useEffect } from 'react';
 import 'antd/dist/antd.css';
 import { useSelector, useDispatch } from 'react-redux';
 import { Routes, Route } from 'react-router-dom';
-import { LoginButton, SignupButton } from './components/Forms';
 import { checkUser } from './redux/features/auth/authSlice';
 import UserInfo from './components/UserInfo';
 import ProfileTabs from './components/ProfileTabs';
+import Navbar from './components/Navbar/Navbar';
```
Comment (Raghad-Mezied) [inline client/src/App.js:8]: Name the file index.jsx

File: client/src/components/Navbar/Navbar.jsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+const { Header } = Layout;
+
+const Navbar = () => {
+  const menu = (
+    <Menu
+      items={[
+        {
+          label: (
+            // must link to profile page
```
Comment (Raghad-Mezied) [inline client/src/components/Navbar/Navbar.jsx:21]: The profile route is ready, you can navigate to it.

File: client/src/components/Navbar/Navbar.jsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        {
+          label: (
+          // must link to logout
+            <Menu.Item onClick={() => console.log(2)} className="logout">
+              <LogoutOutlined className="icon" />
+              Logout
+            </Menu.Item>
+          ),
+        },
+      ]}
```
Comment (Raghad-Mezied) [inline client/src/components/Navbar/Navbar.jsx:37]: Destructure the Item

File: client/src/components/Navbar/Navbar.jsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+          <div className="left">
+            {!user && <SignupButton />}
+            {!user && <LoginButton />}
+            {user && (
+            <Dropdown className="drop" overlay={menu} trigger={['click']} placement="bottom">
+              {/* Must link profile pic to src */}
+              <Avatar src="https://joeschmoe.io/api/v1/random" />
+            </Dropdown>
+            )}
```
Comment (Raghad-Mezied) [inline client/src/components/Navbar/Navbar.jsx:78]: You can render them in one condition

File: client/src/components/Navbar/Navbar.jsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                </Button>
+                <Button className="btn">
+                  Team
+                </Button>
+                <Button className="btn">
+                  Challenge
+                </Button>
+                <Button className="btn">
+                  About
+                </Button>
```
Comment (Raghad-Mezied) [inline client/src/components/Navbar/Navbar.jsx:65]: Why you are using buttons here? check the menu component

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

File: client/src/components/Navbar/index.jsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+            {!user ? (
+              <div>
+                <SignupButton />
+                <LoginButton />
+              </div>
+            ) : (
+              <Dropdown className="drop" overlay={menu} trigger={['click']} placement="bottom">
+                {/* Must link profile pic to src */}
+                <Avatar src="https://joeschmoe.io/api/v1/random" />
+              </Dropdown>
```
Comment (Raghad-Mezied) [inline client/src/components/Navbar/index.jsx:82]: Send an image from the back in checkAuth route and use it here.

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

File: client/src/components/ActiveMembers/index.js [hunk 7]
```diff
@@ -14,6 +14,7 @@ const ActiveMembers = () => {
         const { data: { data } } = await axios.get('/api/users?role=interviewer&limit=4', {
           cancelToken: cancelToken.token,
         });
+        console.log(data);
```
Comment (Raghad-Mezied) [inline client/src/components/ActiveMembers/index.js:17]: Remove the console.

File: .eslintrc.json [hunk 8]
```diff
@@ -15,6 +15,7 @@
         "@typescript-eslint"
     ],
     "rules": {
+        "camelcase": "off",
```
Comment (Raghad-Mezied) [inline .eslintrc.json:18]: Remove this role and rename your variable with camelcase

File: client/src/components/Navbar/index.jsx [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+import { useSelector } from 'react-redux';
+import './style.css';
+import { Link } from 'react-router-dom';
+import logo from '../../assets/images/logo.png';
+import { LoginButton, SignupButton } from '../Forms';
+
+const { Header } = Layout;
+const { Item } = Menu;
+const Navbar = () => {
+  const { user } = useSelector((state) => state.auth);
```
Comment (Raghad-Mezied) [inline client/src/components/Navbar/index.jsx:17]: Destructure the id and image from the user and you can check the if there is a user from isAuthenticated,

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by Raghad-Mezied — APPROVED** [review]

**Review by manar-abed — APPROVED** [review]

**Review by abedshamia — APPROVED** [review]

**Pull request merged**